Sources/RTMP/RTMPConnection.swift

```diff
@@ -455,12 +457,26 @@ open class RTMPConnection: EventDispatcher {
         previousTotalBytesIn = totalBytesIn
         previousTotalBytesOut = totalBytesOut
         previousQueueBytesOut.append(socket.queueBytesOut.value)
+        
+        
+        if currentBytesOutPerSecond == 0 {
+            zeroBytesOutCounter += 1
+            // If currentBytesOutPerSecond has been 0 for the last 5 seconds
+            if zeroBytesOutCounter >= 5 {
+                logger.warn("STALE connection, force close")
+                self.close(isDisconnected: true) // call close function
+                zeroBytesOutCounter = 0 // reset counter
+            }
+        } else {
+            zeroBytesOutCounter = 0 // reset counter
+        }
+        
         for stream in streams {
             stream.on(timer: timer)
         }
         if measureInterval <= previousQueueBytesOut.count {
             var total = 0
-            for i in 0..<previousQueueBytesOut.count - 1 where previousQueueBytesOut[i] < previousQueueBytesOut[i + 1] {
+            for i in 0..<previousQueueBytesOut.count - 1 where previousQueueBytesOut[i] < previousQueueBytesOut[i + 1] && previousQueueBytesOut[i + 1] != 0 {
```

Comment:
> there's a bug that sends publishSufficientBWOccured event

You are absolutely right! Thanks.

Reply:
This value should originally represent the unsent data size for a period of 1 second, so it seemed reasonable to consider it as sufficient when it's 0. However, due to it being synonymous with the size of the most recently enqueued data, issues were occurring. This has been fixed in this pull request: https://github.com/shogo4405/HaishinKit.swift/pull/1264.